### PR TITLE
Attempt to fix flaky Storetest

### PR DESCRIPTION
**Pull request by thuningxu**

### CODE REVIEW

File: services/storehost/storehost_test.go [hunk 1]
```diff
@@ -38,6 +39,20 @@ import (
 	"github.com/uber/cherami-thrift/.generated/go/store"
 )
 
+func timedWgWait(wg *sync.WaitGroup, timeout time.Duration) bool {
```
Comment (KiranRG) [inline services/storehost/storehost_test.go]: "common.AwaitGroup" already does exactly this .. if you'd like to use it.

File: services/storehost/storehost_test.go [hunk 2]
```diff
@@ -69,8 +84,8 @@ func (s *StoreHostSuite) TestStoreHostTimerQueueWriteWithRead() {
 		numExtents = 256                    // number of concurrent extents
 		minMessages, maxMessages = 0, 1000  // messages to send on an extent
 		minMsgSize, maxMsgSize = 512, 2048  // size of message
-		minMsgDelay, maxMsgDelay = 5, 300   // delay to use on messages (in seconds)
-		minRandWait, maxRandWait = 0, 10000 // wait between messages published (in milliseconds)
+		minMsgDelay, maxMsgDelay = 5, 30   // delay to use on messages (in seconds)
+		minRandWait, maxRandWait = 0, 1000 // wait between messages published (in milliseconds)
```
Comment (KiranRG) [inline services/storehost/storehost_test.go]: these values were intended to apply to a super-long test that needs to be initiated manually .. (using the "-long" flag).

File: services/storehost/storehost_test.go [hunk 3]
```diff
@@ -85,10 +100,9 @@ func (s *StoreHostSuite) TestStoreHostTimerQueueWriteWithRead() {
 		var statsLatency = make([]*stats, numExtents)
 		var statsDrift = make([]*stats, numExtents)
 
+		wgAll.Add(numExtents)
+		var readerTimeout int32
 		for x := 0; x < numExtents; x++ {
-
-			wgAll.Add(1)
-
```
Comment (KiranRG) [inline services/storehost/storehost_test.go]: this is okay .. but just thought i'd let you know, this is a pattern i tend to use .. where the wait-group add immediately precedes the go-func call; and the first thing the go-func does is a 'defer waitgroup-done'.

Comment (thuningxu) [inline services/storehost/storehost_test.go]: Honestly, I am also not sure why this makes the tests stable.

**Comment by thuningxu** [comment]
@KiranRG Note that I also applied same logic to `TestStoreHostTimerQueueWriteThenRead()`.

**Pull request merged**